functions-python/preprocessed_analytics/src/main.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from datetime import datetime
+
+import flask
+import functions_framework
+from flask import Response
+
+from helpers.logger import Logger
+from .processors.gbfs_analytics_processor import GBFSAnalyticsProcessor
+from .processors.gtfs_analytics_processor import GTFSAnalyticsProcessor
+
+logging.basicConfig(level=logging.INFO)
+
+
+def get_compute_date(request: flask.Request) -> datetime:
+    """
+    Get the compute date from the request JSON.
+    """
+    try:
+        json_request = request.get_json()
+        compute_date_str = json_request.get("compute_date", None)
+        if compute_date_str:
+            return datetime.strptime(compute_date_str, "%Y%m%d")
+    except Exception as e:
+        logging.error(f"Error getting compute date: {e}")
+    return datetime.now()
+
+
+def process_analytics(request: flask.Request, processor_class) -> Response:
```

Comment:
Please educate me: function_config.json says `process_analytics` is the entry point, but @functions_framework.http is set on process_analytics_gtfs and process_analytics_gbfs. How does it work?

Reply:
I agree that it makes things confusing! I changed the name to `preprocess_analytics` to be consistent.